app/src/test/scala/org/alephium/explorer/persistence/queries/ContractQueriesSpec.scala

```diff
@@ -0,0 +1,103 @@
+import slick.jdbc.PostgresProfile.api._
+
+import org.alephium.api.model.ValAddress
+import org.alephium.explorer.AlephiumFutureSpec
+import org.alephium.explorer.GenApiModel._
+import org.alephium.explorer.GenDBModel._
+import org.alephium.explorer.api.model.Pagination
+import org.alephium.explorer.persistence.{DatabaseFixtureForEach, DBRunner}
+import org.alephium.explorer.persistence.model.{CreateSubContractEventEntity, EventEntity}
+import org.alephium.explorer.persistence.queries.ContractQueries
+import org.alephium.explorer.persistence.schema.CreateSubContractEventSchema
+import org.alephium.protocol.model.Address
+
+@SuppressWarnings(
+  Array("org.wartremover.warts.DefaultArguments", "org.wartremover.warts.AsInstanceOf"))
+class ContractQueriesSpec
+    extends AlephiumFutureSpec
+    with DatabaseFixtureForEach
+    with DBRunner
+    with ScalaFutures {
+
+  def createEventGen(contract: Option[Address] = None): Gen[EventEntity] =
+    for {
+      event       <- eventEntityGen
+      parent      <- contract.map(Gen.const).getOrElse(addressGen)
+      subContract <- addressGen
+    } yield {
+      event.copy(
+        contractAddress = CreateSubContractEventEntity.createContractEventAddress,
+        fields = ArraySeq(
+          ValAddress(subContract),
+          ValAddress(parent)
+        )
+      )
+    }
+
+  "Contract Queries" should {
+    "insertSubContractCreation" in {
+      forAll(Gen.nonEmptyListOf(createEventGen())) { events =>
+        run(CreateSubContractEventSchema.table.delete).futureValue
+        run(ContractQueries.insertSubContractCreation(events)).futureValue
+        run(CreateSubContractEventSchema.table.result).futureValue.sortBy(_.timestamp) is events
+          .flatMap(CreateSubContractEventEntity.fromEventEntity)
+          .sortBy(_.timestamp)
+      }
+    }
+
+    "getParentAddressQuery" in {
+      forAll(Gen.nonEmptyListOf(createEventGen())) { events =>
+        run(CreateSubContractEventSchema.table.delete).futureValue
+        run(ContractQueries.insertSubContractCreation(events)).futureValue
+
+        events.flatMap(CreateSubContractEventEntity.fromEventEntity).foreach { event =>
+          run(ContractQueries.getParentAddressQuery(event.subContract)).futureValue is Some(
+            event.contract)
+        }
+
+        run(ContractQueries.getParentAddressQuery(addressGen.sample.get)).futureValue is None
+      }
+    }
+
+    "getSubContractsQuery" in {
+      val parent     = addressGen.sample.get
+      val pagination = Pagination.unsafe(1, 5)
+
+      forAll(Gen.nonEmptyListOf(createEventGen(Some(parent))), Gen.nonEmptyListOf(createEventGen())) {
+        case (events, otherEvents) =>
+          run(CreateSubContractEventSchema.table.delete).futureValue
+          run(ContractQueries.insertSubContractCreation(events ++ otherEvents)).futureValue
+
+          run(ContractQueries.getSubContractsQuery(parent, pagination)).futureValue is events
+            .sortBy(_.timestamp)
```

Comment:
[Nit] Is it possible to test the order inside of the block too?

Reply:
this test have 1 event per block, I'll try to add a test for that :+1: